test/controllers/returns/credits/CancelCreditsClaimControllerSpec.scala

```diff
@@ -71,125 +71,89 @@ class CancelCreditsClaimControllerSpec extends PlaySpec with JourneyActionAnswer
 
   override def beforeEach(): Unit = {
     super.beforeEach()
-    reset(
-      journeyAction,
-      dataRequest,
-      view,
-      navigator,
-      saveFunction
-    )
-    when(journeyAction.apply(any)).thenAnswer(byConvertingFunctionArgumentsToAction)
-    when(journeyAction.async(any)).thenAnswer(byConvertingFunctionArgumentsToFutureAction)
-    when(dataRequest.userAnswers.fill(any[Gettable[Boolean]], any)(any)).thenReturn(form)
-    when(view.apply(any)(any, any)).thenReturn(HtmlFormat.empty)
+    
+    when(journeyAction.apply(any)) thenAnswer byConvertingFunctionArgumentsToAction
+    when(journeyAction.async(any)) thenAnswer byConvertingFunctionArgumentsToFutureAction
+
+    when(view.apply(any)(any, any)) thenReturn Html("the view")
+    when(formProvider.apply()) thenReturn form
+
+    when(navigator.cancelCreditRoute(any)) thenReturn Call(GET, "/next-page")
+    when(cacheConnector.saveUserAnswerFunc(any)(any)) thenReturn saveFunction
+
+    val x = request.userAnswers
+    when(request.userAnswers.remove(any, any)) thenReturn Try(x)
```

Comment:
does this cloud the tests. This will assume that this setting are needed for the submit function when they are not. The same goes for the form setting above.

Reply:
I wonder if maybe a function that set up the mock and called it when it is needed could help.